manifests/router/service.pp

```diff
@@ -11,12 +11,18 @@
     hasrestart => true,
   }
 
-  if $::qpid::router::open_file_limit and $::systemd {
+  if $::qpid::router::open_file_limit != 1 {
+    $ensure_limit = 'present'
+  } else {
+    $ensure_limit = 'absent'
+  }
+
+  if $::systemd {
     systemd::service_limits { 'qdrouterd.service':
+      ensure => $ensure_limit,
       limits => {
         'LimitNOFILE' => $::qpid::router::open_file_limit,
       },
-      notify => Service['qdrouterd'],
```

Comment:
I'm leaning to keeping this in and set `restart_service` to false. We will need to chain `Class['systemd::systemctl::daemon_reload'] -> Service['qdrouterd']` but the advantage is that we dont have a separate `systemctl restart qdrouterd` and a potential restart through the `Service`.

Reply:
What is the advantage of this vs. the builtin method that systemd:system_limits provides?

Reply:
The built in method can have a double restart: `exec { 'systemctl restart ...': }` + `service { '...': }`. The advantage of theirs is that you don't have to manage the service resource but if you do, the double restart is not worth it IMHO.